### PR TITLE
Add a new note a day after adding the first product

### LINKED ISSUE #5956: New message: How to customize your product catalog
**Title:** How to customize your product catalog
**Copy:** You want your product catalog and images to look great and align with your brand. This guide will give you all the tips you need to get your products looking great in your store.
**Segmentation:** new users
**Timing:** 1 day after adding the first product
**Cadence:** Once
**CTA leads to:** https://docs.woocommerce.com/document/woocommerce-customizer/?utm_source=inbox
**CTA label:** Learn more

### PR DESCRIPTION
**Pull request by moon0326**
Fixes #5956 

This PR adds a new note with a link to the customizer doc a day after adding the first product.

### Screenshots

Screen Shot 2021-01-08 at 10 15 42 AM

### Detailed test instructions:

Make sure your store is less than 14 days old. If it has been active for more than 14 days, run the following query.

```update wp_options set option_value = UNIX_TIMESTAMP() where option_name='woocommerce_admin_install_timestamp'```

1. Start with a fresh installation of WooCommerce or delete all published products.
2. Add a new product.
3. Open `wc_posts` and find the product you just added in step #2. Update the `post_date` and `post_date_gmt` to the past date (at least -1 day)
2. Install & activate "WP Crontrol" plugin
3. Navigate to Tools -> Cron Events and run `wc_admin_daily` job
4. Navigate to WooCommerce -> Home and confirm a new note with the title "How to customize your product catalog"

### CODE REVIEW

**Review by adrianduffell — DISMISSED** [review]
Overall the code looks good 👍  Some tests for `get_note()` would be nice however.

I also found the note is triggered for established stores with existing products, but the advice in the note seems unwanted for those stores.

Perhaps there should be another check to determine if the store is new or established? I think you could use the store’s order count to do this. For example, if the store has existing orders, then don't display the note.

**Comment by moon0326** [comment]
> Overall the code looks good 👍 Some tests for `get_note()` would be nice however.
> 
> I also found the note is triggered for established stores with existing products, but the advice in the note seems unwanted for those stores.
> 
> Perhaps there should be another check to determine if the store is new or established? I think you could use the store’s order count to do this. For example, if the store has existing orders, then don't display the note.

Thank you for catching it 👍  I think that should be fixed. I will push a fix with a test.

I'm also going to see if this PR can use the Remote Inbox Notification system instead.

**Comment by moon0326** [comment]
@adrianduffell Hey Adrian, I asked Pedro about the existing store issue. This is his reply: https://github.com/woocommerce/woocommerce-admin/issues/5956#issuecomment-760085082

tl;dr -- the store is active for less than 14 days.

I've updated the logic and added tests.

File: src/Events.php
Comment (louwie17) [inline src/Events.php:149]: I believe this isn't necessary, as it's already being done above where `possible_add_notes` is being triggered.

https://github.com/woocommerce/woocommerce-admin/pull/6032/files#diff-2e1f1cedde54c64bfd2d2d1b61019c13af57b9c5e1dc1315d4c2dcc492ad66e1R96-R99

File: tests/notes/class-wc-tests-customizing-product-catalog.php
Comment (louwie17) [inline tests/notes/class-wc-tests-customizing-product-catalog.php:1]: Nice work on these test cases, very thorough 👍 👍

**Review by louwie17 — CHANGES_REQUESTED** [review]
Tested the store being older then 14 day's and the note didn't show 👍 also nice work on the tests.
I left one comment in the `CustomizingProductCatalog.php`, I believe calling the `DataSourcePoller` there isn't necessary (see comment).

File: src/Events.php
Comment (moon0326) [reply src/Events.php:149]: Thank you @louwie17 I'm not sure why that those lines are highlighted as new. The last four lines are existing code for the Remote Inbox Engine. I think we can ignore them.

File: tests/notes/class-wc-tests-customizing-product-catalog.php
Comment (moon0326) [reply tests/notes/class-wc-tests-customizing-product-catalog.php:1]: @louwie17 thank you!

File: src/Events.php
Comment (louwie17) [reply src/Events.php:149]: @moon0326 are you sure? they don't show up on the `main` branch: https://github.com/woocommerce/woocommerce-admin/blob/main/src/Events.php#L142

Comment (louwie17) [reply src/Events.php]: Thanks @moon0326, approving this :)

**Review by louwie17 — APPROVED** [review]
LGTM 🚀

**Comment by moon0326** [comment]
> LGTM 🚀

Thank you! I need your approval please 😄

**Comment by louwie17** [comment]
> Thank you! I need your approval please 😄

Ooh shoot haha you might need @adrianduffell approval, or you could dismiss his review, if you can (I haven't run into this before):
Screen Shot 2021-01-20 at 10 03 06 PM

**Comment by moon0326** [comment]
> > Thank you! I need your approval please 😄
> 
> Ooh shoot haha you might need @adrianduffell approval, or you could dismiss his review, if you can (I haven't run into this before):
> Screen Shot 2021-01-20 at 10 03 06 PM

Oh nice! Did not know that was a feature :) Thank you!

**Pull request merged**
